### PR TITLE
Fix WPML Integration meta manipulation

### PR DESCRIPTION
**Pull request by lucatume**
[ECP-1429](https://theeventscalendar.atlassian.net/browse/ECP-1429)

Artifact: [screencast](https://share.cleanshot.com/cmNgCGc0)

Required by [ECP PR](https://github.com/the-events-calendar/events-pro/pull/2212)

The `Tribe__Events__Integrations__WPML__Meta::translate_post_id` mehtod will filter the `get_post_metadata` calls to translate the post IDs that might be found in the meta (i.e. the `_EventOrganizerID` and the `_EventVenueID` ones).
Currently, the filtering will be applied in the `get_metadata_raw` and `metadata_exists` functions and, in both instances, it's for a bail check: one where returning a non-null value would short-circuit the function logic and return the value.  If not found, the value returned should be the original one, `null` in the case of the bail checks, to let the rest of the function logic run its course.

[ECP-1429]: https://theeventscalendar.atlassian.net/browse/ECP-1429?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tribe/Integrations/WPML/Meta.php [hunk 1]
```diff
@@ -90,7 +96,7 @@ public function translate_post_id( $value, $object_id, $meta_key, $single ) {
 
 		$cache[ $cache_key ] = $value;
 
-		return $value;
+		return $single ? $value[0] : $value;
```
Comment (Camwyn) [inline src/Tribe/Integrations/WPML/Meta.php:99]: We probably want to store the amended value (`$value[0]`) in cache - as we directly return it above (line 52) which could lead to unexpected issues if it's an array.

**Review by Camwyn — CHANGES_REQUESTED** [review]
careful with cache 😉

File: src/Tribe/Integrations/WPML/Meta.php (hunk 1 above)
Comment (lucatume) [reply src/Tribe/Integrations/WPML/Meta.php:101]: Fixed.=

**Comment by lucatume** [comment]
I'm debugging the test failure in #4157

**Comment by lucatume** [comment]
I've "fixed" the failing test by skipping it: it passes locally, it passes in CI when running alone ([link](https://github.com/the-events-calendar/the-events-calendar/actions/runs/4306285337/jobs/7509887773)) will `SIGSEV` when running in CI ([link](https://github.com/the-events-calendar/the-events-calendar/actions/runs/4306118295/jobs/7510296677#step:21:28)).
I'm trying to debug the issue in another PR ([link](https://github.com/the-events-calendar/the-events-calendar/pull/4157)).

**Pull request merged**